exercises/week 3/solutions/josei.sh

```diff
@@ -0,0 +1,147 @@
+
+echo "Confirm the balance by mining a few more blocks."
+bitcoin-cli generatetoaddress 1 $(bitcoin-cli -rpcwallet=Miner getnewaddress) > /dev/null
+echo "--------------------------------------------------"
+
+echo "Print the final balances of Alice and Bob."
+echo "Alice's balance: " 
+bitcoin-cli -rpcwallet=Alice getbalance
+echo "Bob's balance: " 
+bitcoin-cli -rpcwallet=Bob getbalance
+echo "--------------------------------------------------"
+
+echo " "
+
+echo "--------------------------------------------------Settle Multisig--------------------------------------------------"
+echo "Create a PSBT to spend funds from the multisig, ensuring 10 BTC is equally distributed back between Alice and Bob after accounting for fees."
+bitcoin-cli -rpcwallet=Alice importaddress $multisigAddress
+bitcoin-cli -rpcwallet=Bob importaddress $multisigAddress
+
+#Get Multisig Address TXID and Vout to use as input in PSBT
+multisigTXID=$(bitcoin-cli -rpcwallet=Alice listunspent|jq -r '.[0]|.txid')
+multisigVout=$(bitcoin-cli -rpcwallet=Alice listunspent|jq -r '.[0]|.vout')
+#echo "New Multisig TXID and Vout: "
+#echo $multisigTXID
+#echo $multisigVout
+
+#Get addresses for Alice and Bob to receive PSBT
+aliceReturnAddress=$(bitcoin-cli -rpcwallet=Alice getnewaddress "Alice return address" )
+bobReturnAddress=$(bitcoin-cli -rpcwallet=Bob getnewaddress "Bob return address" )
+#echo "New addresses for Alice and Bob"
+#echo $aliceReturnAddress
+#echo $bobReturnAddress
+
+#Create new PSBT TX
+newpsbt=$(bitcoin-cli -named createpsbt inputs='''[ { "txid": "'$multisigTXID'", "vout": '$multisigVout' } ]''' outputs='''{ "'$aliceReturnAddress'": 9.999998, "'$bobReturnAddress'": 9.999998 }''')
+
+
+#Instead of individually signing the psbt, chain the signing process.
+#
+#    Sign by Alice first. Get the signed .psbt.
+#    Use the same psbt and sign it by BOB.
+
+echo "#Get Alice to sign"
+aliceNewPSBT=$(bitcoin-cli -rpcwallet=Alice walletprocesspsbt $newpsbt | jq -r '.psbt')
+bitcoin-cli analyzepsbt $aliceNewPSBT
+echo "#Get Bob to sign"
+bothNewPSBT=$(bitcoin-cli -rpcwallet=Bob walletprocesspsbt $aliceNewPSBT | jq -r '.psbt')
+
+
+#
+#In the above process no need to join the two psbts again. Bob's signed psbt output will be the final psbt.
+#
+
+#echo "#Join signatures"
+#newSignedPSBT=$(bitcoin-cli combinepsbt '''["'$aliceNewPSBT'", "'$bobNewPSBT'"]''')
+#bitcoin-cli decodepsbt $newSignedPSBT
+#bitcoin-cli analyzepsbt $newSignedPSBT
+bitcoin-cli analyzepsbt $bothNewPSBT
+echo "--------------------------------------------------"
+echo "#Finalize and Send New Signed PSBT" 
+echo "--------------------------------------------------"
+newSignedPSBTHex=$(bitcoin-cli finalizepsbt $bothNewPSBT | jq -r '.hex')
+
+
+#
+#do not use the wallet here. Just bitcoin-cli sendrawtransaction. For some reason using wallet will fail to broadcast.
+#
+#bitcoin-cli -rpcwallet=Alice sendrawtransaction hexstring=$newSignedPSBTHex
+bitcoin-cli sendrawtransaction hexstring=$newSignedPSBTHex
```

Comment:
last one modification. This will make it work. 

```suggestion
bitcoin-cli sendrawtransaction $newSignedPSBTHex
```